crypto/property/properties.ebnf

```diff
@@ -0,0 +1,17 @@
+(* https://bottlecaps.de/rr/ui *)
+
+Definition
+      ::= PropertyName ( '=' Value )? ( ',' PropertyName ( '=' Value )? )*
+Query ::= ( '-'? PropertyName | PropertyName ( '=' | '!=' ) Value )
+          ( ',' ( '-'? PropertyName | PropertyName ( '=' | '!=' ) Value ) )*
+Value    ::= NumberLiteral
+        | StringLiteral
+StringLiteral ::= QuotedString | UnquotedString
+QuotedString     ::= '"' [^"]* '"'
```

Comment:
I think you should allow backslash-escape inside the quoted string.

Reply:
Why?  so we can write `fips="Y\e\s"` ?

I'd be happy to ban \ inside a string to provide a capability to add it painlessly in the future.

Reply:
Banning it for future expansion helps.  But I meant to only escape quotes, as in things like "Hello \"world\""

Reply:
If anyone is ever going to want to have a Windows path in a quoted string, we'd better *not* use \ as the escape character.

That being said, the \ is more or less universally known as an escape character, and is already used as such elsewhere in OpenSSL, so it's actually a little surprising to have something different for property values.

(and I'm curious, where was ^ inspired from?  That's actually used in VMS file specs, where you'll see things like `openssl-1^.0^.2q^.tar.gz`, but I've not seen you as VMS savvy, so...)

Reply:
> where was ^ inspired from?

The caret at the beginning of a character class inverts the set, i.e. `[^a-b]` is the complement of `[a-b]`. If you want to include a literal caret, it must not be first in the set. See GREP(1).

Reply:
(_Of course_ you know that, it's part of your Perl-fu. It was just too early in the morning... ;-) ).

Reply:
D'oh.

And still, that was after the first coffee.

Reply:
I think the ^ is stretching eBNF a little.